cylc/flow/workflow_files.py

```diff
@@ -1036,18 +1004,149 @@ def get_platforms_from_db(run_dir):
         pri_dao.close()
 
 
-def validate_flow_name(flow_name: str) -> None:
+@overload
+def parse_reg(
+    reg: str, src: Literal[False] = False, require_flow_file: bool = True
+) -> str:
+    ...
+
+
+@overload
+def parse_reg(
+    reg: str, src: Literal[True], require_flow_file: bool = True
+) -> Tuple[str, Path]:
+    ...
+
+
+@overload
+def parse_reg(
+    reg: str, src: bool, require_flow_file: bool
+) -> Union[str, Tuple[str, Path]]:
+    ...  # Need this 3rd overload https://github.com/python/mypy/issues/6113
+
+
+def parse_reg(
+    reg: str, src: bool = False, require_flow_file: bool = True
+) -> Union[str, Tuple[str, Path]]:
+    """Centralised parsing of the workflow argument, to be used by most
+    cylc commands (script modules).
+
+    Infers the latest numbered run if a specific one is not given (e.g.
+    foo -> foo/run3, foo/runN -> foo/run3).
+
+    "Offline" commands (e.g. cylc validate) can usually be used on
+    workflow sources so will need src = True and require_flow_file = True.
+
+    "Online" commands (e.g. cylc stop) are usually only used on workflows in
+    the cylc-run dir so will need src = False.
+
+    Args:
+        reg: The workflow arg. Can be one of:
+            - relative path to the run dir from ~/cylc-run;
+            - absolute path to a run dir, source dir or workflow file (only
+                if src is True);
+            - '.' for the current directory (only if src is True).
+        src: Whether the workflow arg can be a workflow source (i.e. an
+            absolute path (which might not be in ~/cylc-run) and/or a
+            flow.cylc file (or any file really), or '.' for cwd).
+        require_flow_file: (Ignored if src is False, or if reg is already a
+            path to a file.) Whether a flow.cylc (or suite.rc) file is required
+            to be present in the path looked up from the workflow arg.
```

Comment:
For some reason I thought this extra argument would be needed, but having gone through all the scripts I didn't find a single one where you need `src=True, require_flow_file=False`. @oliver-sanders might this be useful in the future or should I just get rid of it?